packages/frontend/src/features/queryRunner/semanticViewerPivotQueries.ts

```diff
@@ -108,9 +102,19 @@ export const getPivotQueryFunctionForSemanticViewer = (
               }
             : undefined;
 
-        const valuesColumns = pivotedResults.columns.filter(
-            (col) => !query.pivot?.index.includes(col),
-        );
+        const valuesColumns = pivotedResults.columns.reduce((acc, col) => {
+            if (!query.pivot?.index.includes(col)) {
+                // TODO: returning a bit of a dummy object here to keep this working,
+                // but the extra information isn't used in the Semantic viewer
+                acc.push({
+                    referenceField: col,
+                    id: col,
+                    aggregation: VizAggregationOptions.ANY,
+                    pivotValues: [],
+                });
+            }
+            return acc;
+        }, [] as PivotValuesColumn[]);
```

Comment:
```ts
const valuesColumns: PivotValuesColumn[] = pivotedResults.columns.reduce<PivotValuesColumn[]>((acc, col) => {
    if (!query.pivot?.index.includes(col)) {
        // Create a new object for the current column
        const newColumn: PivotValuesColumn = {
            referenceField: col,
            id: col,
            aggregation: VizAggregationOptions.ANY,
            pivotValues: [],
        };

        // Use the spread operator to create a new array with the new column
        return [...acc, newColumn];
    }

    return acc;
}, []);
```

a bit safer alternative to avoid `as` and `push`

Reply:
I was going to suggest making this optional or somehow trimming this information from the result metadata for the semantic viewer, but since this can be helpful in the semantic viewer and does not add a lot of data transfer, I am okay to include this metadata for sem viewer as well.